core/trino-spi/src/main/java/io/trino/spi/connector/ColumnBasicMetadata.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+import io.trino.spi.type.Type;
+
+import java.util.Objects;
+
+import static io.trino.spi.connector.SchemaUtil.checkNotEmpty;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+public class ColumnBasicMetadata
+{
+    private final String name;
+    private final Type type;
+    private final boolean nullable;
+    private final boolean hidden;
+
+    public ColumnBasicMetadata(String name, Type type, boolean nullable, boolean hidden)
+    {
+        checkNotEmpty(name, "name");
+        requireNonNull(type, "type is null");
+
+        this.name = name.toLowerCase(ENGLISH);
+        this.type = type;
+        this.hidden = hidden;
+        this.nullable = nullable;
+    }
+
+    public String getName()
+    {
+        return name;
+    }
+
+    public Type getType()
+    {
+        return type;
+    }
+
+    public boolean isNullable()
+    {
+        return nullable;
+    }
+
+    public boolean isHidden()
+    {
+        return hidden;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        ColumnBasicMetadata that = (ColumnBasicMetadata) o;
+        return nullable == that.nullable && hidden == that.hidden && name.equals(that.name) && type.equals(that.type);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(name, type, nullable, hidden);
+    }
+
+    @Override
+    public String toString()
+    {
+        StringBuilder sb = new StringBuilder("ColumnBasicMetadata{");
+        sb.append("name='").append(name).append('\'');
+        sb.append(", type=").append(type);
+        sb.append(", nullable=").append(nullable);
+        sb.append(", hidden=").append(hidden);
+        sb.append('}');
```

Comment:
My IDE generates toString using Joiner.

Reply:
I wanted to be in line with `io.trino.spi.connector.ColumnMetadata#toString`. My IDE allows me to choose how to generate `toString`.

Is there any difference? If so then it would be nice to update other `toString` methods in SPI.